### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,10 +1,52 @@
...
+
+    for player_name, information in players_dict.items():
+
+        race_name = information["race"]["name"]
+        race_description = information["race"]["description"]
+        if not Race.objects.filter(name=race_name).exists():
+            Race.objects.create(
+                name=race_name,
+                description=race_description
+            )
```
Comment (Grayder0152) [inline main.py:20]: Instead of this use [get_or_create](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#get-or-create)

**Review by Grayder0152 — CHANGES_REQUESTED** [review]

File: main.py [hunk 2]
```diff
@@ -1,10 +1,49 @@
...
+            Guild.objects.get_or_create(
+                name=guild_name,
+                description=guild_description
+            )
+
+        Player.objects.create(
+            nickname=player_name,
+            email=information["email"],
+            bio=information["bio"],
+            race=Race.objects.get(name=race_name),
```
Comment (ArsenPidhoretskyi) [inline main.py:42]: Why you need to get it you can do in:
race, _ = Race.objects.get_or_create(
            name=race_name,
            description=race_description
        )

Comment (ArsenPidhoretskyi) [reply main.py:42]: The same with guild

File: main.py [hunk 3]
```diff
@@ -1,10 +1,49 @@
...
+        race_name = information["race"]["name"]
+        race_description = information["race"]["description"]
+        Race.objects.get_or_create(
+            name=race_name,
+            description=race_description
+        )
+        for skill in information["race"]["skills"]:
+            skill_name = skill["name"]
+            skill_bonus = skill["bonus"]
+            skill_race = Race.objects.get(name=race_name)
```
Comment (ArsenPidhoretskyi) [inline main.py:23]: Why you need to get it inside loop get_or_create return instance

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: main.py [hunk 4]
```diff
@@ -1,10 +1,41 @@
...
+                name=information["guild"]["name"],
+                description=information["guild"]["description"]
+            )
+
+        Player.objects.create(
+            nickname=player_name,
+            email=information["email"],
+            bio=information["bio"],
+            race=race,
+            guild=guild if guild is not None else None
```
Comment (ArsenPidhoretskyi) [inline main.py:37]:
```suggestion
            guild=guild
```
This check is so strange if guild is None place None if no place guild

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**